torchvision/csrc/models/modelsimpl.cpp

```diff
@@ -0,0 +1,61 @@
+#include "modelsimpl.h"
+
+namespace vision
+{
+namespace models
+{
+namespace modelsimpl
+{
+ReluImpl::ReluImpl(bool inplace) : inplace(inplace) {}
```

Comment:
We actually explicitly decided not to have them, because they can be used with `torch::nn::Functional` and the respective methods, like `torch::max_pool2d`. This is something I discussed with Zeming at the time, basically we figured that having a module for everything, especially stateless things like functions, is not optimal, even in PyTorch.

So @ShahriarSS, I don't think we should have this file. To e.g. use `ReLU` in a model, write:

```cpp
torch::nn::Functional(torch::relu)
```

and for `MaxPool`:

```
torch::nn::Functional(torch::max_pool2d)
```

Reply:
I actually made a note about this in [the tutorial](https://pytorch.org/tutorials/advanced/cpp_frontend.html):

![screen shot 2019-02-12 at 9 11 50 am](https://user-images.githubusercontent.com/6429851/52654181-4cd53200-2ea6-11e9-94f4-ef42afb07638.png)

Reply:
Oh I haven't seen this tutorial. It must be new. I will change the code accordingly.

Reply:
@goldsborough 
I wont be able to use torch::nn::Functional(torch::relu_) here. Should I wrap relu_ in a function?

Reply:
@goldsborough 
Another problem is that apparently default arguments must be passed to Functional too. For example this doesn't compile:
``` c++
torch::nn::Functional(torch::max_pool2d, 3, 2)
```
But this does:
``` c++
torch::nn::Functional(torch::max_pool2d, 3, 2, 0, 1, false)
```
Is there a better option?

Reply:
Hmm I feel like that should work. Let me look into this.

Why can't yo use `relu_`?

Reply:
@goldsborough 
Because Functional with relu_ won't compile. The problem must be with its return type that is non-const reference or its arguement which is also a non-const reference.

Reply:
@goldsborough I think this part is the problem:
note:   no known conversion for argument 1 from ‘at::Tensor&(at::Tensor&)’ to ‘torch::nn::FunctionalImpl::Function’ {aka ‘std::function<at::Tensor(at::Tensor)>’}

Reply:
@goldsborough The problem with max_pool2d starts here:
```
static_assert(sizeof...(_BoundArgs) == sizeof...(_Args)
```